Forta-Bots-Challenge-3/src/network.ts

```diff
@@ -0,0 +1,81 @@
+import { providers, ethers, Contract, BigNumber } from "ethers";
+import { getEthersProvider } from "forta-agent";
+import {
+  ARBITRUM_DAI_ADDRESS,
+  L1_ARBITRUM_ESCROW_ADDRESS,
+  L1_OPTIMISM_ESCROW_ADDRESS,
+  OPTIMISM_DAI_ADDRESS,
+  ERC20_ABI,
+} from "./constants";
+
+export interface NetworkData {
+  chainId: number;
+  chainName: string;
+  daiContract: string;
+  escrowContract: string;
+  provider: any;
+}
+
+const OPTIMISM_DAI_DATA: NetworkData = {
+  chainId: 10,
+  chainName: "Optimism",
+  daiContract: OPTIMISM_DAI_ADDRESS,
+  escrowContract: L1_OPTIMISM_ESCROW_ADDRESS,
+  provider: new ethers.providers.JsonRpcProvider("https://mainnet.optimism.io"),
+};
+
+const ARBITRUM_DAI_DATA: NetworkData = {
+  chainId: 42161,
+  chainName: "Arbitrum",
+  daiContract: ARBITRUM_DAI_ADDRESS,
+  escrowContract: L1_ARBITRUM_ESCROW_ADDRESS,
+  provider: new ethers.providers.JsonRpcProvider("https://arb1.arbitrum.io/rpc"),
+};
+
+export const NETWORK_MAP: Record<number, NetworkData> = {
+  10: OPTIMISM_DAI_DATA,
+  42161: ARBITRUM_DAI_DATA,
+};
+
+export default class NetworkManager implements NetworkData {
```

Comment:
To piggyback off of @Vxatz's comment, if implemented correctly, there should be no need to implement your own `NetworkManager` as the `GAM` has a `NetworkManager` you could use. [Here](https://github.com/NethermindEth/general-agents-module?tab=readme-ov-file#networkmanager), for quick access.